app/plugin_api.go

```diff
@@ -422,6 +422,30 @@ func (api *PluginAPI) GetEmojiImage(emojiId string) ([]byte, string, *model.AppE
 	return api.app.GetEmojiImage(emojiId)
 }
 
+// Plugin Section
+
+func (api *PluginAPI) GetPlugins() (*model.PluginsResponse, *model.AppError) {
```

Comment:
Hmm I wonder if it would be better to just return a slice of plugin manifests here rather than use `model.PluginsResponse`. If you want plugin statuses you can use `GetPluginStatuses` instead.

Another consideration would be we have this return a list of all plugin manifests, remove `GetPluginStatuses` and just add `GetPluginStatus` so that a plugin can pull that status for only the plugins it cares about. Thoughts @hanzei @crspeller ?

Reply:
Ya I think it would be better to go with @jwilander suggestion.